src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const batchTxn = env.jt(
+                batch::outer(alice, 0, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq),
+                ticket::use(aliceTicketSeq));
+            auto const noopTxn =
+                env.jt(noop(alice), ticket::use(aliceTicketSeq + 1));
+            env(batchTxn);
+            env(noopTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "Payment", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+    }
+
+    void
+    testObjectsOpenLedger(FeatureBitset features)
+    {
+        testcase("objects open ledger");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+
+        // Before Batch Txn
+        {
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            std::uint32_t aliceTicketSeq{env.seq(alice) + 1};
+            env(ticket::create(alice, 10));
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            uint256 const chkId{getCheckIndex(alice, aliceSeq)};
+            auto const batchTxn = env.jt(
+                batch::outer(alice, 0, batchFee, tfAllOrNothing),
+                batch::inner(check::create(alice, bob, XRP(10)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+                ticket::use(aliceTicketSeq));
+            auto const objTxn = env.jt(check::cash(bob, chkId, XRP(10)));
+            env(objTxn, ter(tecNO_ENTRY));
+            env(batchTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "CheckCreate", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+                {3,
+                 "CheckCash",
+                 "tesSUCCESS",
```

Comment:
I don't quite follow why this transaction succeeds/is after the Batch transaction. Is it because it's technically queued but gets let out of the queue very quickly?